⚠️ Outdated attempt for a version 3.0.2

moonglum opened the pull request:
Re-energized by Node finally supporting real compatibility between ESM and CommonJS, I wanted to see how much work it is to update everything to the latest versions. Doesn't seem to be a lot:

* I adjusted the test output, as browser support for newer features has gotten better (who knew!)
* Then I removed mocha, as we can use the built-in test runner with no code changes
* Updated the dependencies to the latest versions with one exception (all without any code changes)
    * The updates also remove the deprecation warnings you see in the version of faucet-js that is currently released
* Updated the node support matrix (still supporting v18 in this release to make it non-breaking. Furthermore, it is still supported until April and there is no reason not to do it)

The one dependency I did not update to the latest version is `@rollup/plugin-commonjs`. Version 27 changes the output in a weird way.

Then I cherry-picked two more changes from the swc branch:
* Drop support for undocumented module format names (`es` and `es6`)
* Bugfix: `NAMELESS_MODULES` used `es` instead of `esm`

moonglum committed:
dropped support for undocumented module-format aliases

moonglum committed:
updated expected transpilation results

moonglum committed:
replaced Mocha with Node's built-in test runner

moonglum committed:
updated dependencies

moonglum committed:
dropped support for obsolete versions of Node

moonglum committed:
fixed nameless module designations

it appears this was never correct in the first place

FND commented on lib/bundle/config.js:
```diff
@@ -8,15 +8,13 @@ let { nodeResolve } = require("@rollup/plugin-node-resolve");
 let MODULE_FORMATS = { // maps faucet identifiers to Rollup identifiers
 	esm: true,
-	es: "esm", // alias
-	es6: "esm", // alias
```
I honestly have no idea why we had those aliases in the first place; "esm" probably wasn't firmly established back then? 👴

moonglum replied:
Yes, exactly

FND commented on lib/bundle/config.js:
```diff
@@ -8,15 +8,13 @@ let { nodeResolve } = require("@rollup/plugin-node-resolve");
 	iife: true
 };
-let NAMELESS_MODULES = new Set(["es", "amd", "cjs"]); // NB: Rollup identifiers
+let NAMELESS_MODULES = new Set(["esm", "amd", "cjs"]); // NB: Rollup identifiers
```
Just to be sure: This should always have been `"esm"` as far as Rollup (not faucet) was concerned?

moonglum replied:
This is my understanding, yes!

FND commented on package.json:
```diff
@@ -19,17 +19,17 @@
 		"test": "npm-run-all lint --parallel test:unit test:cli",
 		"test:cli": "./test/cli/run",
-		"test:unit": "mocha test/unit/test_*.js",
+		"test:unit": "node --test test/unit/test_*.js",
```
This is pretty cool - thank you for caring!

Just to be safe: Does this actually execute the same amount of tests? (I'm afraid I can't check myself at this time.)

moonglum replied:
Yes, it executes all tests. I also made a test fail to make sure the exit code is correct. Will fixup this change :+1:

FND commented on test/unit/test_bundling.js:
```diff
@@ -1,5 +1,5 @@
-/* global describe, it, beforeEach, afterEach */
 "use strict";
+const { describe, it, beforeEach, afterEach } = require("node:test");
 
```
For consistency:

```suggestion

let { describe, it, beforeEach, afterEach } = require("node:test");
```

In fact, strictly speaking, this line should be moved a little further down: Just above `require("path")` (because local > third-party > stdlib).

moonglum replied:
Fix uped :heavy_check_mark:

FND reviewed (approved):
Looks pretty good; excellent work, thank you!

I've only looked at the diff/commits so far, i.e. haven't actually executed any code yet - but will have to do all that anyway as part of the release process (which always scared me a little, even back when I was still familiar with those shenanigans... ).

moonglum commented:
@FND Adressed all your feedback, and force pushed the change :+1: Let me know if anything is missing.

FND commented on test/unit/test_bundling.js:
```diff
@@ -1,8 +1,7 @@
-
 let { MockAssetManager, makeBundle, FIXTURES_DIR } = require("./util");
 let faucetJS = require("../../lib").plugin;
+let { describe, it, beforeEach, afterEach } = require("node:test");
```
If there were pedants around, they might point out that `path` and `assert` should probably use the `node:` prefix as well - fortunately, we don't have to worry about that. Who wants to open that can of worms anyway?

FND replied:
I fixed this while I was at it...

FND reviewed (approved):
LGTM :shipit:

FND committed:
removed npm-run-all dependency

as suggested by @moonglum: this appears to be unmaintained, but it's not
really worth the encumbrance anyway

FND committed:
v3.0.2

FND commented:
@moonglum: I've added a few commits ...

* 560c75c6c6b720d0e72536680739dea56b4e1a15 removed npm-run-all dependency
* 7780cf27dafbf6a7db83d78d3097aa86536d7d41 v3.0.2

Please note the newly added change log (very similar to https://github.com/faucet-pipeline/faucet-pipeline-core/pull/147).

Failing tests are expected because we're referencing pre-release versions (due to the weirdness of our DIY monorepo here).

I've also allowed myself to rewrite history while I was reviewing them for the change log, mostly to adjust commit messages' grammar for consistency.

moonglum commented:
Thanks for getting this ready for release :+1: No objections to making this a 3.0.2 release instead of a 3.1 release. npm-run-all is definitely not a reason against it, as it is a dev dependency. So why would anyone depending on this package care? The only reason would be dropping support for Node 18. But I'm okay with that.

moonglum commented:
Great, then I think this one is ready for release :+1:

FND committed:
updated internals for compatibility with faucet-core v3

note that we don't actually _need_ faucet-core v3 though because `await`
doesn't care whether there's a promise or not

FND commented:
@moonglum: updated to account for faucet-core v3 see c880cac6f9bb8186c9b7965cff4f49a02bbded24 - though we'll have to update the change log after deciding on a version number

moonglum commented:
Thanks! I would probably still set to core v3 only. We need to make sure that all pipelines use the same version of faucet, otherwise things break. I would therefore set them all to require core 3.

moonglum commented on CHANGELOG.md:
```diff
@@ -2,6 +2,20 @@ faucet-pipeline-js version history
 ==================================
 
 
+v3.0.2
```
I think we will go for 3.1.0, right?

moonglum commented:
Closed in favor of #236
